src/NServiceBus.Core/SendContext.cs

```diff
@@ -0,0 +1,102 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Allows the users to control how the send is performed
+    /// </summary>
+    public class SendContext
```

Comment:
in the interest of consistency I would like to see some numbers on the impact of the usage of classes versus structs in the pipeline. then based on those numbers do a pull request that reflect that information. ATM a class in this context is in keeping with the majority of the code base and a pull request of this nature should not diverge from that consistency